pokedex/network/netcore.lua

```diff
@@ -0,0 +1,948 @@
+local tcp_server = require "defnet.tcp_server"
+local tcp_client = require "defnet.tcp_client"
+local ljson = require "defsave.json"
+local profiles = require "pokedex.profiles"
+local p2p_discovery = require "pokedex.network.p2p"
+local notify = require "utils.notify"
+local broadcast = require "utils.broadcast"
+local settings = require "pokedex.settings"
+local log = require "utils.log"
+
+local M = {}
+
+M.MSG_NEARBY_SERVER_FOUND = hash("netcore_nearby_server_found")
+M.MSG_STATE_CHANGED = hash("netcore_state_changed")
+
+M.STATE_FINAL = hash("final")
+M.STATE_IDLE = hash("idle")
+M.STATE_SERVING = hash("serving")
+M.STATE_CONNECTING = hash("connecting")
+M.STATE_CONNECTED = hash("connected")
+
+M.DEFAULT_HOST_PORT = 9120
+
+local current_state = M.STATE_IDLE
+local is_listening = false
+
+local server
+local version
```

Comment:
This is causing it to try to concat with nil further down. Looks like it should be the app version? I think this should be fine
`local version = sys.get_config("project.version"))`